python/triton/runtime/interpreter.py

```diff
@@ -1069,7 +1089,10 @@ def _init_args_hst(self, args_dev, kwargs):
     def _restore_args_dev(self, args_dev, args_hst, kwargs, kwargs_hst):
         for arg_dev, arg_hst in zip(args_dev, args_hst):
             if hasattr(arg_dev, "data_ptr"):
-                arg_dev.data.copy_(arg_hst.to(arg_dev.device).data)
+                # No need to rewrap because this just modifies internal
+                arg_dev, arg_hst = _unwrap_tensor(arg_dev), _unwrap_tensor(arg_hst)
+                arg_dev.untyped_storage().copy_(arg_hst.untyped_storage().to(device=arg_dev.device))
```

Comment:
NIT: Cross-device copy is supported for tensor, I presume this should work for storages as well?
```suggestion
                arg_dev.untyped_storage().copy_(arg_hst.untyped_storage())
```

Reply:
I tested it and it worked!